### ISSUE TITLE
hidden dropout w/ 'predict' not performing as expected

### ISSUE THREAD

**Issue by white54503** [lmjohns3/theanets#100]
I'm looking to extract model predictive mean, variance and precision from a recurrent regressor model by calling 'hidden_dropout' as both training regularizer and uncertainty estimator, a la Yarin Gal's Monte Carlo Dropout.  My implementation trains and predicts without error but fails to produce the sort of Monte Carlo prognostication described by Gal.  

A code snippet for reference:

```
#Define and instantiate the network
net=[]
layer0 = dict(form='mut1', size=256*3)
net = theanets.recurrent.Regressor([num_feats,layer0,layer0,layer0, 1])

#Training
losses = []
BATCH_SIZE = 16
for tm, _ in net.itertrain([train_feats[:,155:,:],train_sigs[:,155:,:]],
                           [valid_feats[:,155:,:],valid_sigs[:,155:,:]],
                           monitor_gradients=False,
                           batch_size=BATCH_SIZE,
                           algorithm='adam',
                           patience=15,
                           hidden_dropout=0.30,
                           #input_dropout=0.3,
                           weight_l2=1e-4,
                           learning_rate=1e-4,
                           #momentum=0.3,
                           validate_every=10,
                           min_improvement=0.00001):
    losses.append(tm['loss'])
    
#Prediction  
prd0 = net.predict(valid_feats[0:1,155:,:],hidden_dropout=0.3)
prd1 = net.predict(valid_feats[0:1,155:,:],hidden_dropout=0.5)
prd2 = net.predict(valid_feats[0:1,155:,:],hidden_dropout=0.7)
prd3 = net.predict(valid_feats[0:1,155:,:],hidden_dropout=0.99)
```

**Comment by lmjohns3** [comment]
Thanks for reporting this! It turned out that the dropout regularizer was not being applied properly because of a bug in my understanding of how python's ``fnmatch`` module works. I fixed it up and dropout seems to be working properly again in ``train()`` and ``predict()``.
